mozci/push.py

```diff
@@ -341,7 +343,83 @@ def add(result):
 
             normalized_tasks.append(task)
 
-        return [Task.create(**task) for task in normalized_tasks]
+        tasks = [Task.create(**task) for task in normalized_tasks]
+        # Pass tasks instead of trying to access self.tasks which would make this recursive
+        tasks = self.fetch_errors_results(tasks)
+        return tasks
+
+    def fetch_errors_results(self, tasks: list) -> list:
+        """ Fetch errors and results for test tasks. Define ADR_CONFIG_PATH and adr.cache for the caching to work"""
+        test_tasks = adr.config.cache.get(self.push_uuid, {})
+        if len(test_tasks.keys()) > 0:
+            # Let's add cached error summaries to TestTasks
+            for t in tasks:
+                error_summary = test_tasks.get(t.id)
+                # Only Test tasks have stored error summary information in the cache
+                if error_summary:
+                    t._errors = error_summary["errors"]
+                    t._results = error_summary["results"]
+            logger.info("Fetched tasks for {} from the cache".format(self.push_uuid))
+        else:
+            # This list will make reference to tasks that don't have the results or errors properties
```

Comment:
Instead of doing this, I think it would be easier (and a lot less code would be required) to just avoid caching in the ADR query and then always cache ourselves everything.

Reply:
I can add the parameter to avoid the caching (that got lost when the 3rd `run_query` got dropped), however, ADR will fetch that data regardless if we say it not to cache it.

In other words, ADR already got some of the data for us so instead of fetching it again via Taskcluster we simply cache it.

Reply:
Yeah, we should check the cache first and if it's in the cache avoid running the query (which would also fix my other comment https://github.com/mozilla/mozci/pull/159#issuecomment-623461999).